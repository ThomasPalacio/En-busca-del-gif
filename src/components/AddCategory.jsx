
import PropTypes  from 'prop-types';
import React, { useState } from 'react';



export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) =>{
        const {value} = target;
        setInputValue(value)
    }

    const onSubmit = (e) =>{
        e.preventDefault(); 
        const newInputValue = inputValue.trim();

        if (newInputValue.length <= 1) return;

        onNewCategory(newInputValue);
        setInputValue('');
    }

    
    return (
        <form onSubmit={onSubmit} aria-label='form'>
            <input 
                type='text'
                placeholder='Buscar Gif'
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
