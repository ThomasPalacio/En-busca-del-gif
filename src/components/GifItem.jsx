import PropTypes  from "prop-types"



export const GifItem = ({title,url}) => {
  return (
    <div className="card">
        <img src={url} alt={title}  className="card-img-top"/>
        <div className="card-body d-flex justify-content-center align-items-center">
          <p className="card-title">{title}</p>
        </div>
    </div>
    )
}

GifItem.propTypes = {
  title:PropTypes.string.isRequired,
  url:PropTypes.string.isRequired
}