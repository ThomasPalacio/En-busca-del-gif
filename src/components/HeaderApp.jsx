

export const HeaderApp = () => {
  return (
    <div className="d-flex justify-content-center">
        <div>
          <h1 className="textTitles">Buscador de GIF'S </h1>
          <h4 className="subtitleHeader">Descubr√≠ tu favorito en un instante.</h4>
        </div>
    </div>

  )
}
