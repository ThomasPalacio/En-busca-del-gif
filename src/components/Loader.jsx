
import { Dimmer, Loader, Image, Segment } from 'semantic-ui-react'

export const LoaderFunction = () => (
  <div className='container d-flex justify-content-center align-items-center'>
    <Segment>
      {/* <Dimmer active>
        <Loader>Loading</Loader>
      </Dimmer> */}

      <Image src='https://react.semantic-ui.com/images/wireframe/short-paragraph.png' />
    </Segment>
  </div>
)
