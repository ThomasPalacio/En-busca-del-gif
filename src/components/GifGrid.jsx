import { GifItem } from "./GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs";
import { LoaderFunction } from "./Loader";
import PropTypes  from "prop-types";



export const GifGrid = ({category}) => {
const categoryUpper = category.toUpperCase();
const {images,isLoading} = useFetchGifs(category);


  return (
    <div className="mt-5">
      
      <h2 className="textTitles d-flex align-item-center"><i className="fi fi-ss-bullet"></i>{categoryUpper}</h2>
      {
        isLoading && (<LoaderFunction/>)
      }
      
      <div className="card-grid">
      {images.map(({title,url,id}) => 
        <GifItem key={id} 
        title={title}
        url={url}
        />
      )}
      </div>
    </div>
  )
}

GifGrid.propTypes = {
  category:  PropTypes.string.isRequired,
}