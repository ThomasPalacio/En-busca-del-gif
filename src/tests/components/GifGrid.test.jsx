import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs');

describe('Prueba en <GifGrid/>', () => {
    const category = 'Friends';
    
    test('Debe de mostrar el loding inicialmente', () => {
        
        useFetchGifs.mockReturnValue({
            images:[],
            isLoading:true
        })
        render(<GifGrid category={category}/>)
        expect(screen.getAllByRole('img'))
        expect(screen.getByText(category.toUpperCase())) 
    });


    test('Debe de mostar items cuando se cargan las imagenes', () => {
        const gifs = [{
            id:'abc',
            title:'Friends',
            url:'https://addadsa.com'
        },
        {
            id:'abc322',
            title:'Goku',
            url:'https://ssfdsfs.com'
        }
    
    ]
        
        useFetchGifs.mockReturnValue({
            images:gifs,
            isLoading:false
        })
        const gifGrid = render(<GifGrid category={category}/>)
        
        expect(screen.getAllByRole('img').length).toBe(2);
        
        expect(gifGrid).toMatchSnapshot()
    });
});