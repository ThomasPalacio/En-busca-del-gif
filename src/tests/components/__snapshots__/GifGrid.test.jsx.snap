// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGrid/> Debe de mostar items cuando se cargan las imagenes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-5"
      >
        <h2
          class="textTitles d-flex align-item-center"
        >
          <i
            class="fi fi-ss-bullet"
          />
          FRIENDS
        </h2>
        <div
          class="card-grid"
        >
          <div
            class="card"
          >
            <img
              alt="Friends"
              class="card-img-top"
              src="https://addadsa.com"
            />
            <div
              class="card-body d-flex justify-content-center align-items-center"
            >
              <p
                class="card-title"
              >
                Friends
              </p>
            </div>
          </div>
          <div
            class="card"
          >
            <img
              alt="Goku"
              class="card-img-top"
              src="https://ssfdsfs.com"
            />
            <div
              class="card-body d-flex justify-content-center align-items-center"
            >
              <p
                class="card-title"
              >
                Goku
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-5"
    >
      <h2
        class="textTitles d-flex align-item-center"
      >
        <i
          class="fi fi-ss-bullet"
        />
        FRIENDS
      </h2>
      <div
        class="card-grid"
      >
        <div
          class="card"
        >
          <img
            alt="Friends"
            class="card-img-top"
            src="https://addadsa.com"
          />
          <div
            class="card-body d-flex justify-content-center align-items-center"
          >
            <p
              class="card-title"
            >
              Friends
            </p>
          </div>
        </div>
        <div
          class="card"
        >
          <img
            alt="Goku"
            class="card-img-top"
            src="https://ssfdsfs.com"
          />
          <div
            class="card-body d-flex justify-content-center align-items-center"
          >
            <p
              class="card-title"
            >
              Goku
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
